### PR TITLE
Display classroom members.

### PR DESCRIPTION
**Pull request by simoneduca**
TODO: atm the same list of students is displayed for any classroom. FIX IT!

### CODE REVIEW

**Comment by simoneduca** [comment]
@eatyourgreens or @shaunanoordin this is ready for review, thanks.

**Comment by shaunanoordin** [comment]
Hey @simoneduca ! I've encountered a bit of wonkiness:

Issue: after loading a list of classrooms, clicking on a single classroom causes the whole app to 'crash', freezing in place and no longer responding to any user input.

Analysis: Error message
`Uncaught TypeError: Cannot read property 'length' of undefined` (Chrome)
`TypeError: allMembers is undefined` (FF)
at Classroom.jsx,

```
renderStudentList(allMembers, classroomMembers) {
    const list = (allMembers.length > 0) ? allMembers : 'No student found';
```

Hypothesis: if you haven't encountered this error on your end, then this may be specific to certain Zooniverse user accounts; perhaps the fact that I haven't joined any classes causes my allMembers list to be undefined?

Tested on Chrome48/FF44 on MacbookPro+OSX with Privacy Mode.

File: src/containers/Classroom.jsx [hunk 1]
```diff
@@ -8,9 +8,10 @@ import { default as ClassroomPresentational } from '../presentational/Classroom.
 export default class Classroom extends Component {
 
   render() {
+    const included = this.props.classrooms.included;
     const classroom = this.props.classrooms.data.find(classroom =>
       classroom.id === this.props.params.classroomId);
-    return (<ClassroomPresentational data={classroom} />);
+    return (<ClassroomPresentational data={classroom} included={included} />);
```
Comment (shaunanoordin) [inline src/containers/Classroom.jsx]: Minor comment: the code in your update passes the sanity check very well, but the `included` variable isn't quite descriptive of its purpose. "What is `included`?" doesn't seem too clear to me; perhaps a brief //comment would help clarify it?

Comment (simoneduca) [inline src/containers/Classroom.jsx]: Ah yes, good catch!

File: src/presentational/Classroom.jsx [hunk 2]
```diff
@@ -23,15 +24,43 @@ export default class Classroom extends Component {
...
-    let token = nextProps.data.attributes.join_token;
+    let classrooms = nextProps.data;
     this.state = {
-      url: routes.root + routes.students + 'join?id=' + id + '&token=' + token
+      url: routes.root + routes.students + 'join?id=' + classrooms.id + '&token=' + classrooms.attributes.join_token
     };
   }
 
+  renderStudentList(allMembers, classroomMembers) {
+    const list = (allMembers.length > 0) ? allMembers : 'No student found';
```
Comment (eatyourgreens) [inline src/presentational/Classroom.jsx]: This sets `list` to a string if there are no students, which will break any code that depends on it being an array. Maybe set it `[]` instead?

Comment (simoneduca) [inline src/presentational/Classroom.jsx]: Ouch! Good one, thanks! You're right this would break in the case there are no members at all yet.

**Comment by simoneduca** [comment]
Review comments/issues should be sorted now. @shaunanoordin can you confirm you don't get the error anymore?

**Comment by simoneduca** [comment]
Right, should be ready to go now @shaunanoordin

**Comment by shaunanoordin** [comment]
Sweet!
- Previous issue addressed; error no longer detected.
- Teacher->Classroom pages are accessible as usual.
- Code passes sanity check.

Status: **Verified**
Action: **Merging**

**Pull request merged**